src/coreclr/jit/codegenlinear.cpp

```diff
@@ -753,7 +753,13 @@ void CodeGen::genCodeForBBlist()
                 break;
 
             case BBJ_ALWAYS:
-                inst_JMP(EJ_jmp, block->bbJumpDest);
+                inst_JMP(EJ_jmp, block->bbJumpDest,
+#ifdef TARGET_AMD64
+                         /* isJmpAlways */ !GetEmitter()->emitIsLastInsCall() && !block->hasAlign()
+#else
+                         /* isJmpAlways */ false
```

Comment:
Does this mean the optimization is only implemented for x64? If so, why?

Reply:
Because x86 and arm32 are not a priority targets for me and I have no way to test locally on arm64. The original changes should have been portable to arm64 with some minor changes but now we're using idCodeSize() that isn't true because code size doesn't exist on arm.